clients/lcdproc/disk.c

```diff
@@ -124,32 +137,71 @@ disk_screen(int rep, int display, int *flags_ptr)
 	sock_printf(sock, "widget_set D f 1 2 %i %i %i %i v 12\n", lcd_wid, lcd_hgt, lcd_wid, count);
 	for (i = 0; i < count; i++) {
 		char tmp[lcd_wid + 1];	/* should be large enough */
+		char cap[8] = {'\0'};
+		char dev[256];
+		u_int64_t size = 0;
+		int full = 0;
 
-		if (table[i].dev[0] == '\0')
+		// Eewww, goto... but shy of restructuring everything it's 
+		// the cleanest way to skip the rest of the outer loop. :-/
+		next_mnt:
+
+		if (mnt[i].mpoint[0] == '\0')
 			continue;
 
-		if (i >= num_disks) {	/* Make sure we have enough lines... */
-			sock_printf(sock, "widget_add D s%i string -in f\n", i);
-			sock_printf(sock, "widget_add D h%i hbar -in f\n", i);
+		// Prescreen for entries to ignore.
+		for (int ign_ct=0; ign_ct<ignore_count; ign_ct++) {
+			if (strcmp(mnt[i].mpoint, disk_ignore[ign_ct])==0) {
+				ig_offset++;
+				i++;
+				goto next_mnt; // Behold, the one legitimate use for goto ;-)
+			}
 		}
-		if (lcd_wid >= 20) {	/* 20+x columns */
-			sprintf(tmp, "%-*s %6s E%*sF", dev_wid, table[i].dev, table[i].cap, gauge_wid, "");
+
+		// Parse/format the incoming FS data:
+		int i_widget = i-ig_offset; // Index into the widget array due to "ignore" list
+
```

Comment:
Just having `i_widget` as secondary counter in the loop probably would be a lot simpler.

Reply:
I did manage to remove one of the two new vars :+1:  Don't think we can get away from maintaining an offset though